source/AI.cpp

```diff
@@ -308,7 +349,126 @@ AI::AI(const List<Ship> &ships, const List<Minable> &minables, const List<Flotsa
 
 
 
+
+
+
+
+// Fleet commands from the player.
+void AI::IssueFormationChange(const PlayerInfo &player)
+{
+	// Figure out what ships we are giving orders to
+	vector<Ship *> targetShips = GetShipsForFormationCommand(player);
+	
+	// This should never happen, but just in case:
+	if(targetShips.empty())
+		return;
+	
+	const auto &formationPatterns = GameData::Formations();
+	if(formationPatterns.size() < 1)
```

Comment:
I'll add the `empty` function.

If a formation-pattern only contains invalid statements, then it would be loaded as an empty formation. The FormationPattern would then just return Point(0,0) for any ships requesting positions in such empty formation.
Is this the type of invalid pattern you were thinking of?

Reply:
Yeah.